### DISCUSSION TITLE
Caculation about the iso-curve of Schmid factor in the IPF plot

### DISCUSSION THREAD

**Discussion by slimrabbit001** [mtex-toolbox/mtex#991] [Ask Anything]
HELLO,
My name is David,a Ph.D student in Materials engineering , I'm recently working on the study of plasticity of Ti6Al4V to reveal the relationship between grain orientation and slip activities ,a example has been attached as below.
![1619449438(1)](https://user-images.githubusercontent.com/83230996/116106636-e0405980-a6e4-11eb-9c63-c97ac71692c8.jpg)
I want to generate such red and blue iso-curves which represent Schmid factor isocontours of basal and prismatic slip systems.However I'm a rookie to MTex,I have no clue for this,please share any idea with me.
Thanks in advance!
David

**Comment by kilir** [comment]
Hi,
I guess Ralfs comment here https://github.com/mtex-toolbox/mtex/discussions/919#discussioncomment-551121 should answer your question in the broadest, general sense.
Cheers, Rüdiger

**Reply by slimrabbit001** [reply]
Hi Rüdiger
MUCH THANKS to your reply! By referring to Ralfs' comments,i actually generate a simliar iso-curve triangle map.It's not quiet complex,i just copy the code fro, the Mtex helping documents about Schmid factor calculation ,and combine it with 'd = Miller(plotS2Grid(cs.fundamentalSector),cs)',and it's work!! MUCH THANKS again for your help!
Best wishes!
David

**Comment by QPF-BS** [comment]
Hi, I'am working on the study of Ni-based alloy. i also want to make the Schmid factor isocontours on IPOF. BUT I could not use theMTEX

**Reply by King-wang-H** [reply]
Hi,I am also learning to draw such a diagram recently.And  I'am working on the study of Mg alloy. Have you solved your problem?
I wonder if it is convenient for you to have your advice.

**Reply by slimrabbit001** [reply]
Well you can try the code i ploted in the commons

**Comment by QPF-BS** [comment]
Mr. Ralf Hielscher had made it at  Pole Figure, But how to make it at an IPF?
![image](https://user-images.githubusercontent.com/84628290/119229414-8ea3b700-bb4a-11eb-8d8d-a654e541f4aa.png)

**Reply by slimrabbit001** [reply]
HELLO
I used the 'cs.fundamentalSector' ,I don't understand the actual meanings but it just work.here is my complete code i wrote,still there is a problem that i can't solve while if you have idea share with me please: i can't lable the iso-curve value on the curve,i used the commands 'showTexts',but it's not working.
```matlab
cs = crystalSymmetry('6/mmm');

d = Miller(plotS2Grid(cs.fundamentalSector),cs);
n_1 = Miller(1,0,-1,0,cs,'hkil');
d_1 = Miller(1,-2,1,0,cs,'uvtw');
sS_1 = slipSystem(n_1,d_1)
sSAll_1 = sS_1.symmetrise('antipodal')
tau_1=sSAll_1.SchmidFactor(d);
[tauMax_1,id] = max(abs(tau_1),[],2);
contourf(d,tauMax_1,'contourf',0.4:0.02:0.55,'grid','grid_res',10*degree,'ShowText','on','LineColor','b','Fill','off');hold all
```
Wish it could help!
David

**Reply by QPF-BS** [reply]
I replaced the alloy with Ni, and the program reported an error

**Reply by slimrabbit001** [reply]
Could you post the error report ?the code should works fine,maybe you Can check your slip plane and slip direction to make sure they are orthogonal

**Reply by QPF-BS** [reply]
```matlab
cs = crystalSymmetry('m-3m');

d = Miller(plotS2Grid(cs.fundamentalSector),cs);
n_1 = Miller(1,1,1,cs,'hkil');
d_1 = Miller(0,-1,1, cs,'uvw');
sS_1 = slipSystem(n_1,d_1)
sSAll_1 = sS_1.symmetrise('antipodal')
tau_1=sSAll_1.SchmidFactor(d);
[tauMax_1,id] = max(abs(tau_1),[],2);
contourf(d,tauMax_1,'contourf',0.4:0.02:0.55,'grid','grid_res',10*degree,'ShowText','on','LineColor','b','Fill','off');hold all
```

**Reply by slimrabbit001** [reply]
Try replace  hkil  to hkl，then it should work

**Reply by zmichels** [reply]
I tried the code as it is (no change) and it worked fine for me.
![image](https://user-images.githubusercontent.com/8769619/119262370-91c99200-bba0-11eb-979d-095862725ac5.png)

**Reply by QPF-BS** [reply]
yeah，thank you, I try again, and it worked fine
